pkg/minikube/cluster/cluster_test.go

```diff
@@ -147,6 +149,48 @@ func TestStartHostExists(t *testing.T) {
 	}
 }
 
+func TestStartHostErrMachineNotExist(t *testing.T) {
+	RegisterMockDriver(t)
+	api := tests.NewMockAPI(t)
+	// Create an incomplete host with machine does not exist error(i.e. User Interrupt Cancel)
+	api.NotExistError = true
+	ih, err := createHost(api, defaultMachineConfig)
+	if err != nil {
+		t.Fatalf("Error creating host: %v", err)
+	}
+
+	md := &tests.MockDetector{Provisioner: &tests.MockProvisioner{}}
+	provision.SetDetector(md)
+
+	mc := defaultMachineConfig
+	mc.Name = ih.Name
+
+	// This should pass with creating host, while machine does not exist.
+	ih, err = StartHost(api, mc)
```

Comment:
how about StartHost returns its own ErrorMachineNotFound so we dont depend on virtualbox and it is extensible for other drivers?

Reply:
I added new function `isMachineExists`.
This func checks all driver's MachineDoesNotError.